skyvern/forge/sdk/artifact/storage/base.py

```diff
@@ -10,6 +10,8 @@
     ArtifactType.SCREENSHOT_LLM: "png",
     ArtifactType.SCREENSHOT_ACTION: "png",
     ArtifactType.SCREENSHOT_FINAL: "png",
+    ArtifactType.SKYVERN_LOG: "log",
```

Comment:
Step log..

Task log..

Workflow log...

Reply:
i don't think adding STEP_LOG, TASK_LOG, WORKFLOW_LOG helps.

It's better to step_id, task_id and workflow_run_id columns to filter artifacts. Also, step log and task log are not mutually exclusive. TaskLogs should have the StepLogs. We don't have to create 2 sets of log artifacts that have duplicated data.

Reply:
I'm not sure if I agree with this -- why not have multiple artifact types that may or may not include duplicate data? It makes the implementation + writing of data + reading of data dead simple. 

Why is it better to use step_id / task_id / workflow_run_id columns to filter artifacts?

Reply:
If I understand correctly in this case we'll store workflow and task level logs in the step artifact folders

Current implementation of `bulid` uri always assumes step:
```py
def build_uri(self, artifact_id: str, step: Step, artifact_type: ArtifactType) -> str:
        file_ext = FILE_EXTENTSION_MAP[artifact_type]
        return f"file://{self.artifact_path}/{step.task_id}/{step.order:02d}_{step.retry_index}_{step.step_id}/{datetime.utcnow().isoformat()}_{artifact_id}_{artifact_type}.{file_ext}"
```

We could define a build uri method for logs that would take entity type and id

So that the uri would look like this:
```py
return f"file://{self.artifact_path}/{entity_type}_{entity_id}/{datetime.utcnow().isoformat()}_{artifact_type}.{file_ext}"
```

This way we would be able to fetch logs on the frontend using a similar query as for the other artifacts

We would be able to use artifact type for all the levels

Reply:
@suchintan no concern of data duplication. you're wright, having more artifact_types makes the implementation easier.